src/runtime_src/tools/xclbinutil/XclBinUtilities.cxx

```diff
@@ -1007,10 +1010,144 @@ XclBinUtilities::createMemoryBankGrouping(XclBin & xclbin)
+  //   construct the string for sections to be removed, add to container A
+  //   construct PSD for the sections in <temp>/<index name>/transform, add to container B
+  // }
+  // for each section in container A
+  //   removeSection(PSD)
+  // for each PSD in container B
+  //   addSection(PSD)
+  // delete the temp dir
+
+  // create a temp directory
+  // fs::current_path or fs::temp_directory_path?
+  std::string apJson = "aie_partition.json";
+  // ap: aie_partition
+  fs::path tempDir = fs::current_path() / "ap_temp";
+
+  std::vector<std::string> removeSections;
+  std::vector<std::string> addSections;
+  std::vector<Section*> sections = xclbin.findSection(AIE_PARTITION, true);
+  for (auto& pSection : sections) {
+    // get the section name, index
+    std::string sSectionKind = pSection->getSectionKindAsString();
+    std::string sSectionIndex = pSection->getSectionIndexName();
+    std::string sSectionName = pSection->getName();
+    // std::cout << "sSectionKind = " << sSectionKind << std::endl;
+    // std::cout << "sSectionIndex = " << sSectionIndex << std::endl;
+    // std::cout << "sSectionName = " << sSectionName << std::endl;
+
+    // note sSectionIndex could be an empty string
+    fs::path origDir = tempDir / sSectionIndex / "orig";
+    fs::path transformDir = tempDir / sSectionIndex / "transform";
+    fs::path origApJsonPath = origDir / apJson;
+    fs::path tranApJsonPath = transformDir / apJson;
+    // std::cout << "origDir = " << origDir.string() << std::endl;
+
+    try {
+      fs::create_directories(origDir);
+    }
+    catch (std::exception& e) { // Not using fs::filesystem_error since std::bad_alloc can throw too.
+        std::cout << e.what() << std::endl;
+    }
+    // std::cout << "Temporary directory created: " << origDir << std::endl;
+
+    // construct the PSD for dumpЅection
+    std::string sDumpSection = sSectionKind + "[" + sSectionIndex + "]:JSON:" + origApJsonPath.string();
+    // std::cout << "sDumpSection = " << sDumpSection << std::endl;
+    ParameterSectionData dumpPsd(sDumpSection);
+    xclbin.dumpSection(dumpPsd);
+
+    std::string sRemoveSection = sSectionKind + "[" + sSectionIndex + "]";
+    removeSections.push_back(sRemoveSection);
+
+    // after dumpSection, <temp>/<index name>/orig/ should be populated with pdi files
+    // copy it to <temp>/<index name>/transform/
+    fs::copy(origDir, transformDir);
+
+    // transform the pdi in the transform/ folder
+    // Iterate over the files in the directory
+    for (const auto& entry : fs::directory_iterator(transformDir)) {
+      // Check if the current entry is a regular file and matches the extension
+      if (fs::is_regular_file(entry) && entry.path().extension() == ".pdi") {
+        // std::cout << "pdi file found: " << entry.path() << std::endl;
+
+        if (transform_PDI_file (entry.path().string())) {
+          std::cout << "pdi file transform failed: " << entry.path() << std::endl;
```

Comment:
And so what?

Reply:
Replaced std::system() with the exiting utility function XUtil::exec() which throws on error